File: mobile/lib/data/models/restaurant.dart
class Restaurant {
  const Restaurant({
    required this.id,
    required this.name,
    required this.imageUrl,
    required this.rating,
    required this.deliveryTimeMinutes,
    required this.deliveryFee,
    required this.cuisineType,
    required this.isOpen,
  });

  final int id;
  final String name;
  final String imageUrl;
  final double rating;
  final int deliveryTimeMinutes;
  final double deliveryFee;
  final String cuisineType;
  final bool isOpen;

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['id'] as int,
      name: (json['name'] ?? '') as String,
      imageUrl: (json['image_url'] ?? '') as String,
      rating: (json['rating'] as num?)?.toDouble() ?? 0,
      deliveryTimeMinutes: (json['delivery_time_minutes'] as num?)?.toInt() ?? 0,
      deliveryFee: (json['delivery_fee'] as num?)?.toDouble() ?? 0,
      cuisineType: (json['cuisine_type'] ?? '') as String,
      isOpen: (json['is_open'] as bool?) ?? false,
    );
  }
}

File: backend/models/restaurant.py
from sqlalchemy import Boolean, Float, ForeignKey, Integer, String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from database import Base


class Restaurant(Base):
    __tablename__ = "restaurants"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    image_url: Mapped[str] = mapped_column(String(500), nullable=False)
    rating: Mapped[float] = mapped_column(Float, nullable=False)
    delivery_time_minutes: Mapped[int] = mapped_column(Integer, nullable=False)
    delivery_fee: Mapped[float] = mapped_column(Float, nullable=False)
    cuisine_type: Mapped[str] = mapped_column(String(80), nullable=False)
    is_open: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    menu_items: Mapped[list["MenuItem"]] = relationship(
        "MenuItem", back_populates="restaurant", cascade="all, delete-orphan"
    )


class MenuItem(Base):
    __tablename__ = "menu_items"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, index=True)
    restaurant_id: Mapped[int] = mapped_column(ForeignKey("restaurants.id"), nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(120), nullable=False)
    description: Mapped[str] = mapped_column(String(500), nullable=False)
    category: Mapped[str] = mapped_column(String(80), nullable=False)
    price: Mapped[float] = mapped_column(Float, nullable=False)
    image_url: Mapped[str] = mapped_column(String(500), nullable=False)
    is_available: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)

    restaurant: Mapped[Restaurant] = relationship("Restaurant", back_populates="menu_items")
